### PR TITLE
Get rid of `workvec` in `rotatez.jl`

### PR DESCRIPTION
**Pull request by ZongyuLi-umich**

### CODE REVIEW

**Comment by ZongyuLi-umich** [comment]
Ooops, code coverage failed.

File: src/rotatez.jl [hunk 1]
```diff
@@ -44,107 +48,95 @@ end
...
+        linearinterp!(interp, s, e, len)
         mul!((@view output[:, i]), interp, (@view img[:, i])) # need mul! to avoid allocating
     end
     return output
 end
 
 
 """
-    rotate_x_adj!(output, img, tan_θ, workvec, interp)
+    rotate_x_adj!(output, img, tan_θ, interp)
```
Comment (JeffFessler) [inline src/rotatez.jl:75]: it looks like `rotate_x_adj!` is identical to `rotate_x!` right?
So we do not need both right?

File: src/rotatez.jl [hunk 2]
```diff
@@ -44,107 +48,95 @@ end
...
+        linearinterp!(interp, s, e, len)
         mul!((@view output[i, :]), interp, (@view img[i, :]))
     end
     return output
 end
 
 
 """
-    rotate_y_adj!(output, img, sin_θ, workvec, interp)
+    rotate_y_adj!(output, img, sin_θ, interp)
```
Comment (JeffFessler) [inline src/rotatez.jl:123]: same here?

**Review by JeffFessler — COMMENTED** [review]
Nice simplifications!  See my comments about apparently identical methods?

**Comment by ZongyuLi-umich** [comment]
> I don't understand why github is proposing 26 commits for just 2 changed files. I think something is not ideal about the merging process. I've seen others complain about this online - can you look into it and see if it can be simplified?

Yes, that's a bit annoying, let me take a look.

File: src/rotatez.jl [hunk 3]
```diff
@@ -312,9 +304,9 @@ function imrotate_adj!(
         tan_mod_theta = tan(mod_theta / 2)
         sin_mod_theta = - sin(mod_theta)
         padzero!(plan.workmat1, img, (plan.padsize, plan.padsize, plan.padsize, plan.padsize))
-        rotate_x_adj!(plan.workmat2, plan.workmat1, tan_mod_theta, plan.workvec, plan.interp)
-        rotate_y_adj!(plan.workmat1, plan.workmat2, sin_mod_theta, plan.workvec, plan.interp)
-        rotate_x_adj!(plan.workmat2, plan.workmat1, tan_mod_theta, plan.workvec, plan.interp)
+        rotate_x_adj!(plan.workmat2, plan.workmat1, tan_mod_theta, plan.interp)
+        rotate_y_adj!(plan.workmat1, plan.workmat2, sin_mod_theta, plan.interp)
+        rotate_x_adj!(plan.workmat2, plan.workmat1, tan_mod_theta, plan.interp)
```
[the remark this answers is not in the document]
Comment (ZongyuLi-umich) [reply src/rotatez.jl:309]: linearinterp! won't work for adjoint type of `SparseInterpolator`. It gives me the following error:
`type Adjoint has no field C`

[the remark this answers is not in the document]
Comment (ZongyuLi-umich) [reply src/rotatez.jl:309]: So I will stick on this version of rotation and start implementing `foreach` if `fft_convolve.jl`.

**Pull request closed without merging**